File: testjavabenchmark/presentation/gui/stumbleupon/WallGUI.java
package presentation.gui.stumbleupon;

import domain.stumbleupon.StumbleException;
//Functional requirement 3.2.2
//Test case 165: All classes in package presentation.gui.stumbleupon are not allowed to use in a not direct lower layer
//Result: TRUE
public class WallGUI {
	public WallGUI(){
		try {
			getWall();			
			//FR5.8
		} catch (StumbleException e) {
			e.printStackTrace();
		}
	}

	//FR5.8
	public String getWall() throws StumbleException{
		//FR5.8
		throw new StumbleException("No Wall available");
	}
}

File: testprojects/c#/benchmark/presentation/gui/stumbleupon/WallGUI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CSharpBenchmark.domain.stumbleupon;

namespace CSharpBenchmark.presentation.gui.stumbleupon
{
    //Functional requirement 3.2.2
    //Test case 165: All classes in package presentation.gui.stumbleupon are not allowed to use in a not direct lower layer
    //Result: TRUE
    public class WallGUI
    {
        public WallGUI()
        {
            try
            {
                //FR5.8
            }
            catch (StumbleException e)
            {
                Console.WriteLine(e.StackTrace);
            }
        }
    }
}